### PR TITLE
NIFI-3300 Implemented usage of ZooKeeper chroot capability in the con…

### PR DESCRIPTION
**Pull request by jtstorck**
…nect string

Updated ZooKeeper connect string parsing tests
Updated admin doc for ZooKeeper Migrator migration of nifi root nodes, updated source and destination ZK check by servers in the connection string instead of the entire connection string

Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [x] Have you written or updated unit tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by brosander** [comment]
Reviewing

File: nifi-toolkit/nifi-toolkit-zookeeper-migrator/src/main/java/org/apache/nifi/toolkit/zkmigrator/ZooKeeperEndpointConfig.java [hunk 1]
```diff
@@ -21,24 +21,37 @@
...
 import com.google.common.base.Splitter;
 import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
 
+import java.util.List;
 import java.util.Objects;
 
 class ZooKeeperEndpointConfig {
     private final String connectString;
+    private final List servers;
```
Comment (brosander) [inline nifi-toolkit/nifi-toolkit-zookeeper-migrator/src/main/java/org/apache/nifi/toolkit/zkmigrator/ZooKeeperEndpointConfig.java]: should this be List<String> ?

Comment (jtstorck) [inline nifi-toolkit/nifi-toolkit-zookeeper-migrator/src/main/java/org/apache/nifi/toolkit/zkmigrator/ZooKeeperEndpointConfig.java]: Yes, fixed.

**Comment by brosander** [comment]
@jtstorck I'm hitting an issue restoring to a different path on the same server

```
Exception in thread "main" java.lang.IllegalArgumentException: Source ZooKeeper config ZooKeeperEndpointConfig{connectString=zk-kerberos:2181/, servers=[zk-kerberos:2181], path=/} for the data provided can not be the same as the configured destination ZooKeeper config ZooKeeperEndpointConfig{connectString=zk-kerberos:2181/abcd, servers=[zk-kerberos:2181], path=/abcd}
        at com.google.common.base.Preconditions.checkArgument(Preconditions.java:145)
        at org.apache.nifi.toolkit.zkmigrator.ZooKeeperMigrator.writeZooKeeper(ZooKeeperMigrator.java:137)
        at org.apache.nifi.toolkit.zkmigrator.ZooKeeperMigratorMain.main(ZooKeeperMigratorMain.java:160)
```

**Comment by jtstorck** [comment]
@brosander Good catch, this should be allowed.  I'll update the code and add a test for this use case.

**Comment by brosander** [comment]
+1 Merging

**Pull request closed without merging**